### DISCUSSION TITLE
LSP server not starting automatically

### DISCUSSION THREAD

**Discussion by major-winter** [prabirshrestha/vim-lsp#1327] [Q&A]
Hi, I'm using Neovim with vim-lsp and vim-lsp settings. I'm able to download the lsp, typescript-language-server in this case, but it does not start automatically when I switch between two different filetype. Is there a way to start the lsp manually or any config that would help me to achieve that?

**Comment by mattn** [comment]
Could you please provide step to reproduce your issue?

**Reply by major-winter** [reply]
Hi, I just install the lsp using `:LspInstallServer` command. The server is installed successfully but doesn't start automatically when I enter the file. You can see the picture for reference.
![vim-lsp](https://user-images.githubusercontent.com/54270747/176609066-0115c85c-02a3-40cf-b195-e784d52d6e4f.png)

**Comment by mattn** [comment]
What is your filetype?

**Reply by major-winter** [reply]
it's javascript, and I've already added `let g:lsp_settings_filetype_javascript=['typescript-language-server']` option in my .vimrc.

**Reply by mattn** [reply]
> when I switch between two different filetype

What do you mean ?

**Reply by major-winter** [reply]
I mean that if the `typescript-language-server` is running when I'm in a js file and I switch to another filetype different than js. When I move back to the js file I was working on, the server is not running anymore.

**Reply by mattn** [reply]
AFAIK, vim-lsp and vim-lsp-settings (at least vim-lsp-settings) does not considers changing filetype.

**Reply by mattn** [reply]
But I'm thinking it's better to be supported.

**Comment by Daghall** [comment]
I am using vim 8.2, and I have a similar issue, where files in some folders (git repositories) just would not trigger a server start.

I did some crude debugging and found that for the folders where it did not work, the `server_info["blocklist"]` contained `["typescript","javascript","typescriptreact","javascriptreact"]`, whereas the folders where it works, this list is empty.

All files had `server_info["allowlist"]` equal to `["javascript","javascriptreact","typescript","typescriptreact","typescript.tsx"]]` when opening.

I traced the problem to [vim-lsp-settings for JavaScript/TypeScript](https://github.com/mattn/vim-lsp-settings/blob/dd8ba8ebc55ea3aac3e963483093c1b059cb1fd9/settings/typescript-language-server.vim#L9). It looks like the `blocklist` is populated if there is no `node_moduels` in any parent directory of the active file, so a quick and dirty workaround for me was `mkdir ~/node_modules`. 😅 🙈 🤷
